### ISSUE TITLE
Multiline strings (or block scalars) are not preserved when they have trailing white space.

### ISSUE THREAD

**Issue by bferg88** [mikefarah/yq#566]
**Describe the bug**
Multiline strings (or block scalars) are not preserved when they have trailing white space. They are converted to quoted strings.

version of yq: 3.3.4
operating system: Linux Mint 19

**Input Yaml**
```yaml
this: |
  should 
  really work
```

**Command**
The command you ran:
```
yq merge data.yml
```

**Actual behavior**
```yaml
this: "should \nreally work\n"
```
**Expected behavior**

```yaml
this: |
  should 
  really work
```

**Comment by jaroslawhartman** [comment]
This seems to be affecting Linux version (which is very annoying!). On Mac (checked `darwin_arm`) no issues.

Verified version 4.25.3, `yq_linux_amd64` and `yq_darwin_arm64`

**Comment by Orrimp** [comment]
We have the same issue and its important for us because it destroys the configuration.

**Comment by arikkfir** [comment]
Same here - is this something to be expected in `yq`?

**Comment by arikkfir** [comment]
Ok just discovered this is a duplicate of #1681 and #1277, which both depend on go-yaml/yaml#880 

@mikefarah perhaps it would be best to close #1681 and #1277 as duplicates of this issue (since this one is the oldest)

**Comment by rattboi** [comment]
I spent some time investigating this yesterday, and found the related code within `go-yaml/yaml` that could potentially be changed to fix this. However, I also see that `go-yaml/yaml` currently has 112 PRs that have gone unreviewed, most without comment. 

I almost spent time working out a fix in the upstream library, but I'm wondering if @mikefarah would be open to another direction, i.e. using https://github.com/goccy/go-yaml/ for more of the standard parsing / lexing / encoding / decoding within yq. I see it's already in use for one specific use-case (colorized output), but it could likely be used to replace all `go-yaml/yaml`.

It may be an undertaking, but as upstream of `go-yaml/yaml` seems effectively dead, and benchmarks show https://github.com/goccy/go-yaml/ as being generally 2x faster in operations, it seems like a potential solution for these yaml-decoding issues that are blocked by upstream.

There's a simple `ycat` command bundled with goccy/go-yaml, but it was enough to see how it would handle literal scalars with trailing spaces:

```
echo -ne 'foo: |\n  bar:2    \n  baz:3\n  #foo' | go run ycat.go /dev/stdin
 1 | foo: |
 2 |   bar:2    
 3 |   baz:3
 4 |   #foo
```

**Comment by vit-zikmund** [comment]
Hi there, Mike!

While mr. Niemeyer has [sensible reasons not to allow other maintainers on the project](https://github.com/go-yaml/yaml/issues/788#issuecomment-933888037), in fact the repo [hasn't been touched for 2 years](https://github.com/go-yaml/yaml/commits/v3/) now. The suggested project seems a bit more alive, but in the end it's yet another single-person-team project :( It has a half of opened issues, although that also doesn't have to mean anything and I understand the risk of porting `yq` to it (if that's even a possibility without a major rewrite). But where to go from here? `go-yaml` has a ridiculous number of forks... I guess I know why :face_with_head_bandage: 

I'd say `yq` became so better these days, running on those rusty wheels must be getting challenging to say the least. Not being able to provide any real advice :see_no_evil:, I'm [also](https://github.com/mikefarah/yq/issues/2077#issuecomment-2205554385) curious, whether moving to goccy's `go-yaml` is being considered.

Thank you!

**Comment by pascal-hofmann** [comment]
I use the following as a workaround for now. It removes all trailing spaces from block literals.
```
yq -i '... |= (select(tag == "!!str" and style == "literal") | sub(" *\n", "\n"))' file.yaml
````

**Comment by ingydotnet** [comment]
https://github.com/mikefarah/yq/issues/1781 seems like same issue

**Comment by ingydotnet** [comment]
Also https://github.com/mikefarah/yq/issues/1681
See https://github.com/go-yaml/yaml/issues/880

**Comment by justfalter** [comment]
I've created an issue in yaml/go-yaml to track https://github.com/yaml/go-yaml/issues/78

**Comment by ingydotnet** [comment]
https://github.com/yaml/go-yaml/pull/79 "fixes" this, but I need to discuss it more before merging.

**Comment by ingydotnet** [comment]
@mikefarah re https://github.com/yaml/go-yaml/pull/79#issuecomment-3208952435

From a yq PoV would changing this formatting be considered a breaking change?

I _think_ it's better (at least for yq) because people are starting with yaml with literal scalars and getting literal scalars back on output. But it's still a change in existing behavior.

Keen to hear your take.

**Comment by dieter-adesso** [comment]
> [@mikefarah](https://github.com/mikefarah) re [yaml/go-yaml#79 (comment)](https://github.com/yaml/go-yaml/pull/79#issuecomment-3208952435)
> 
> From a yq PoV would changing this formatting be considered a breaking change?
> 
> I _think_ it's better (at least for yq) because people are starting with yaml with literal scalars and getting literal scalars back on output. But it's still a change in existing behavior.
> 
> Keen to hear your take.

Just hit this bug and it took me longer time to identify the root-cause, because a blank at the end of a line is hard to see.
In my opinion, as long as results are 1:1, it is not a breaking change, because 1:1 output is generated without quoting, if there are no trailing blanks anywhere. But you think probably, that yq is a linter. Maybe you can think about any kind of parameter, that enforces or avoids quoting of such block scalars with trailing blanks.
